controllers/machineset_controller.go

```diff
@@ -249,29 +257,19 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 	}
 
 	syncErr := r.syncReplicas(ctx, machineSet, filteredMachines)
-
-	ms := machineSet.DeepCopy()
-	newStatus, err := r.calculateStatus(ctx, cluster, ms, filteredMachines)
-	if err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "failed to calculate MachineSet's Status")
```

Comment:
In the before diff, we were always calculating the status of the MachineSet after syncing replicas, even if there was a syncErr. With these changes, we now are returning after the error, which is a change in behavior. Was this intentional? If so, what's the reasoning behind it?